hal/armv7r/zynqmp/ddr_init.c

```diff
@@ -0,0 +1,433 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * ZynqMP DDR4 initialization
+ * Based on ZynqMP FSBL code by Xilinx, licensed under MIT license.
+ *
+ * Copyright 2025 Phoenix Systems
+ * Author: Jacek Maksymowicz
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "zynqmp.h"
+#include "zynqmp_regs.h"
+#include "config.h"
+
+#include <board_config.h>
+
+#define DDRC_BASE_ADDRESS    0xfd070000
+#define DDR_PHY_BASE_ADDRESS 0xfd080000
+
+struct {
+	volatile u32 *ddrc;
+	volatile u32 *ddr_phy;
+} ddrInit_common;
+
+static struct {
```

Comment:
otherwise will be copied to .data
```suggestion
static const struct {
```